cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -370,6 +372,22 @@ void addApplication(Id.Application id,
   WorkflowToken getWorkflowToken(Id.Workflow workflowId, String workflowRunId);
 
   /**
+   * Add node state for the given {@link Workflow} run. This method is used to update the
+   * state of the custom actions started by Workflow.
```

Comment:
Yes. Potentially can also be used for Predicates. For programs (MapReduce and Spark) `store.setStart` and `store.setStop` methods are used.